libr/bin/format/objc/mach0_classes.c

```diff
@@ -0,0 +1,734 @@
+static void Mach0_class_(get_class_ro_t)(mach0_ut p, RBinFile *arch,
+										ut32 *is_meta_class,
+										RBinClass *processed_class)
+{
+	Mach0_struct_(SClassRoT) cro;
+	ut64 r;
+	ut32 offset, left;
+
+	r = Mach0_class_(get_pointer)(p, &offset, &left, arch);
+	if(r == 0)
+		return;
+	memset(&cro, '\0', sizeof(Mach0_struct_(SClassRoT)));
+	if(left < sizeof(Mach0_struct_(SClassRoT))){
+		r_buf_read_at(arch->buf, r, (ut8 *)&cro, left);
+	}
+	else {
+		r_buf_read_at(arch->buf, r,
+					(ut8 *)&cro,
+					sizeof(Mach0_struct_(SClassRoT)));
+	}
+
+	r = Mach0_class_(get_pointer)(cro.name, NULL, &left, arch);
+	if(r != 0) {
+		processed_class->name = malloc(left);
+		r_buf_read_at(arch->buf, r, (ut8 *)processed_class->name, left);
+
+//		R_FREE(name);
+	}
+
+	if(cro.baseMethods != 0) {
+		Mach0_class_(get_method_list_t)(cro.baseMethods, arch,
+										processed_class->name,
+										processed_class);
+	}
+
+	if(cro.baseProtocols != 0) {
+		Mach0_class_(get_protocol_list_t)(cro.baseProtocols,
+										arch,
+										processed_class);
+	}
+
+	if(cro.ivars != 0) {
+		Mach0_class_(get_ivar_list_t)(cro.ivars,
+									arch,
+									processed_class);
+	}
+
+	if(cro.baseProperties != 0) {
+		Mach0_class_(get_objc_property_list)(cro.baseProperties,
+											arch,
+											processed_class);
+	}
+
+	if(is_meta_class) {
+		*is_meta_class = (cro.flags & RO_META) ? 1 : 0;
+	}
+}
+
+///////////////////////////////////////////////////////////////////////////////
+static void Mach0_class_(get_class_t)(mach0_ut p, RBinFile *arch,
+									RBinClass *processed_class)
+{
+	Mach0_struct_(SClass) c;
+	mach0_ut r = 0;
+	ut32 offset = 0, left = 0;
+	ut32 is_meta_class = 0;
+
+	r = Mach0_class_(get_pointer)(p, &offset, &left, arch);
+	if (r == 0) {
+		return;
+	}
+	memset(&c, '\0', sizeof(Mach0_struct_(SClass)));
+
+	if (left < sizeof(Mach0_struct_(SClass))) {
+		r_buf_read_at(arch->buf, r, (ut8 *)&c, left);
+	} else {
+		r_buf_read_at(arch->buf, r, (ut8 *)&c, sizeof(Mach0_struct_(SClass)));
+	}
+
+	processed_class->addr = c.isa;
+	Mach0_class_(get_class_ro_t)((c.data) & ~0x3, arch,\
+								&is_meta_class, processed_class);
+}
+
+///////////////////////////////////////////////////////////////////////////////
+static void Mach0_class_(__r_bin_class_free)(RBinClass *p) {
+	RListIter *iter = NULL;
+	RBinSymbol *symbol = NULL;
+	RBinField *field = NULL;
+
+	if (!p) {
+		return;
+	}
+
+	r_list_foreach(p->methods, iter, symbol) {
+		R_FREE(symbol);
+	}
+
+	r_list_foreach(p->fields, iter, field) {
+		R_FREE(field);
+	}
+
+	r_list_free(p->methods);
+	r_list_free(p->fields);
+	r_bin_class_free (p);
+}
+
+RList* Mach0_class_(get_classes)(RBinFile *arch)
+{
+	RList/*<RBinClass>*/ *ret = NULL;
+
+	RBinClass *processed_class = NULL;
+
+	RList *sctns = NULL;
+	RListIter *iter = NULL;
+	RBinSection *s = NULL;
+	ut32 i = 0, size = sizeof(ut32);
+	mach0_ut p;
+	ut32 left;
+	ut8 isFound = 0;
+
+	ut64 numOfUnnamedClass = 0;
```

Comment:
We usually use names like : num_unnamed_class (or something shorter) instead of numOfUnnamedClass